### PR TITLE
frontend: open source shared components package

### PR DESCRIPTION
**Pull request by DeltaMichael**
https://jira.eng.vmware.com/browse/TAUR-5878

### What

Open source shared components library
    
- replace references to taurus, supercollider, vmw, vmware and tms with vdk
- change license to Apache-2.0
- upade copyright headers to reflect license change
- regenerate package-lock.json
- remove e2e tests
- change package versions to 0.0.1
- update zone.js version to fix broken async test
- remove author tag from docs

### How has this been tested

Followed instructions to build and test here https://github.com/vmware/versatile-data-kit/blob/person/mdilyan/os-shared-components/projects/frontend/shared-components/gui/README.md

https://user-images.githubusercontent.com/91800778/218074170-89ea2cca-1676-47b7-9df2-bad81d2b1ff9.mov

### What type of change are you making

New feature, non-breaking

### Follow-up issues

https://github.com/vmware/versatile-data-kit/issues/1623
https://github.com/vmware/versatile-data-kit/issues/1641
https://github.com/vmware/versatile-data-kit/issues/1643
https://github.com/vmware/versatile-data-kit/issues/1644

Signed-off-by: Dilyan Marinov <mdilyan@vmware.com>

### CODE REVIEW

**Comment by DeltaMichael** [comment]
Should we also re-generate package-lock.json? I can't show it here, but it seems to point to http://build-artifactory.eng.vmware.com.

File: projects/frontend/shared-components/gui/.eslintrc.json
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/.eslintrc.json:50]: cypress is for e2e testing, no e2e tests so we may drop that

File: projects/frontend/shared-components/gui/.gitignore copy
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/.gitignore copy:2]: nit

File: projects/frontend/shared-components/gui/.npmrc
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/.npmrc:1]: maybe an additional post-merge PR, to explain in the VEP/README what we have here

File: projects/frontend/shared-components/gui/angular.json
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/angular.json:124]: So we have basically empty documentation-ui screens at this point.
Should we keep those (`documentation-ui/`) - in case we would in the future need to visualise some newly added components, or should we simply not add those until we need to actually visualise something?

In case we decide to keep it, we need to document the environments supported by `documentation-ui`

File: projects/frontend/shared-components/gui/cypress.json
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/cypress.json:9]: no e2e tests to run

File: projects/frontend/shared-components/gui/package.json
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/package.json:3]: what's the version of the wrapper? I think it was 3.0.0

File: projects/frontend/shared-components/gui/package.json
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/package.json:19]: no e2e

File: projects/frontend/shared-components/gui/projects/documentation-ui/.browserslistrc
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/projects/documentation-ui/.browserslistrc:1]: nit: should we link browser support to the VEP

**Review by ivakoleva — COMMENTED** [review]
LVGTM,
need to complete the review of all files, it's a massive PR,
posting what I have so far

File: projects/frontend/shared-components/gui/projects/shared/src/lib/common/predicate/simple/index.ts
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/common/predicate/simple/index.ts:1]: Maybe need to reconfigure `.pre-commit-config.yaml` adding ts support for licensing source file headers?

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/css/utils.scss
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/commons/css/utils.scss:24]: nit: maybe add ticket to resolve those warnings

File: projects/frontend/shared-components/gui/proxy.config.json
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/proxy.config.json:17]: I think we can drop/empty this file entirely, after verification (metadata and lineage are out of scope for the shared components library)

File: projects/frontend/shared-components/gui/sonar-project.properties
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/sonar-project.properties:1]: Most likely we are going to continue using codacy in favour of introducing sonar in parallel. We can drop this file for now, and revise in a dedicated effort for code quality check.

**Review by ivakoleva — COMMENTED** [review]
LGTM,
lets address the source file headers , and have another round of review

**Comment by ivakoleva** [comment]
> Should we be re-generating package-lock.json? I did it because there were lots of packages pointing to 

yes

File: projects/frontend/shared-components/gui/angular.json
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/angular.json:124]: I don't have much context, maybe @tozka will have some valuable input. That being said, I'm generally against introducing code that "we might need some day". It clutters the codebase.

**Comment by DeltaMichael** [comment]
> Should we be re-generating package-lock.json? I did it because there were lots of packages pointing to
> yes, it's re-generated via build from scratch right

Yeah, it's generated on first build if it doesn't exist. However, we're committing it to the repo. Instructions in the README.md also state that we have to run `npm ci --omit=optional`, which tells npm not to try and update packages, but use whatever versions are in package-lock.json. I had to regenerate it, because it was pointing to internal repos. 

Not adding package-lock.json to version control and updating versions on every build is a little bit controversial, so most projects add it.

I was afraid there would be some version conflict when I regenerated it, but everything seems to work fine.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/css/utils.scss
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/css/utils.scss:24]: Not sure what this is. 😅 Are you referring to the static code analysis warnings?

File: projects/frontend/shared-components/gui/projects/shared/src/lib/common/predicate/simple/index.ts
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/common/predicate/simple/index.ts:1]: Done.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/css/utils.scss
Comment (ivakoleva) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/css/utils.scss:24]: Yeah - modifying the code opened in order to handle the quality warnings, is certainly out of scope. 
Could we link to the warnings in a ticket, so we revise those in an additional effort planned?

File: projects/frontend/shared-components/gui/angular.json
Comment (ivakoleva) [reply projects/frontend/shared-components/gui/angular.json:124]: I fancy the simple approach of only including what's needed, so in favour of dropping `documentation-ui/` from here + VEP 1507(in additional PR)

File: projects/frontend/shared-components/gui/projects/shared/package.json
Comment (ivakoleva) [inline projects/frontend/shared-components/gui/projects/shared/package.json:47]: a change of keywords?
I think of adding "versatile-data-kit", "vdk" - "taurus" could be cleared up (is was the name before open-sourcing the component)

File: projects/frontend/shared-components/gui/package.json
Comment (ivakoleva) [reply projects/frontend/shared-components/gui/package.json]: Since we are having a unique artifact prefix `@vdk`, maybe we need to reconsider and start from 1 or something.
Not sure yet if we need version.txt file explicitly

Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/package.json]: I don't think it matter that much, to be honest. It depends on if we consider it a new package or open-sourcing and existing package. But I'm ok with doing either. I've currently set both package.json files to point to version 3.0.0

File: projects/frontend/shared-components/gui/projects/shared/src/lib/common/predicate/simple/index.ts
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/common/predicate/simple/index.ts:1]: Ticket to update docs https://github.com/vmware/versatile-data-kit/issues/1622

**Review by ivakoleva — APPROVED** [review]
I am considering setting the `@vdk` artifacts version to 1, in the scope of the CI/CD prep 😊

I've fetched this PR changes locally, then tested with a (yet local) branch of frontend/data-pipelines/* I am now wrapping up for code review.
(I can confirm the shared library is operational.)

File: projects/frontend/shared-components/gui/projects/documentation-ui/karma.conf.js
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/documentation-ui/karma.conf.js:1]: Generally, yes. For this PR, if it's just configuration files, I'm not sure what the best practice is. Can you really license a configuration?

**Comment by VDK** [comment]
Please untag me

**Comment by DeltaMichael** [comment]
I'll wait for https://github.com/vmware/versatile-data-kit/pull/1625 and https://github.com/vmware/versatile-data-kit/pull/1634 before merging. It seems like it will complicate things if the headers are added and then have to be changed again.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts
Comment (sabadzhiev) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts:11]: Please, remove the internal username references from this and all the files below.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/copy-to-clipboard-button/copy-to-clipboard-button.l10n.ts
Comment (sabadzhiev) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/copy-to-clipboard-button/copy-to-clipboard-button.l10n.ts:1]: We do not support l10n in the applications consuming these components. 

I guess you can get rid of the support here, as well?

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/index.ts
Comment (sabadzhiev) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/index.ts:7]: I guess you can rename the VmwSearchModule to VdkSearchModule as well.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts
Comment (sabadzhiev) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts:69]: Nice commented code :). 

I vote to delete it :).

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.ts
Comment (sabadzhiev) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.ts:19]: also VmwSearchComponent can be VdkSearchComponent. 

In general, check all the class names exported by the project and rename all of them.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/toast/toast.component.ts
Comment (sabadzhiev) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/toast/toast.component.ts:257]: Please, remove the commented code.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-utils/simple-translate-service/index.ts
Comment (sabadzhiev) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-utils/simple-translate-service/index.ts:1]: I believe once the l10n support is removed from the components, this service will be redundant and can be wiped out as well.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/core/navigation/services/navigation.service.ts
Comment (sabadzhiev) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/core/navigation/services/navigation.service.ts:25]: Please, remove TMS, SuperCollider and Control Plane occurrences from the code base. 

All these are VMW internals.

Comment (sabadzhiev) [inline projects/frontend/shared-components/gui/projects/shared/src/lib/core/navigation/services/navigation.service.ts:25]: Please, remove TMS, SuperCollider and Control Plane occurrences from the code base. 

All these are VMW internals.

File: projects/frontend/shared-components/gui/.npmrc
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/.npmrc:1]: This flag reverts to the previous npm behavior of not installing peer deps by default. Not sure why we have this.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts
Comment (ivakoleva) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts:6]: Why?

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/copy-to-clipboard-button/copy-to-clipboard-button.l10n.ts
Comment (ivakoleva) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/copy-to-clipboard-button/copy-to-clipboard-button.l10n.ts]: I agree; those several l10n.ts files were a part of quickly fetched ngx-component sources.
In case we need to, again, fetch updates of those and sync, then we need to keep the differences or restrain from diverging too much from origin.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts
Comment (ivakoleva) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts:64]: nit: in case we decide the several lines of commented code could help us to diverge less (or diverge in a readable/understandable way) - we may keep those

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/toast/toast.component.ts
Comment (ivakoleva) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/toast/toast.component.ts:252]: nit: sane here

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts:64]: In this case, someone commented out a test. I don't think it will help us diverge less. Generally, having commented code is just bad.

Comment (ivakoleva) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts:64]: nit:  did you comment it out, or did the authors? I doubt it was the authors (haven't seen any ngx comment leftovers so far), so it's probably a test that got broekn during transferring.
I vote for deleting it, in case it's not operational. To check if operational, you need to run it, and/or descope it for CI/CD PRs (always could add task comments so we don't forget)

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-utils/simple-translate-service/index.ts
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-utils/simple-translate-service/index.ts]: I'm not messing with l10 support in this PR. It requires me to change too much code as lots of components actually use this service. We can address it as a separate issue.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/copy-to-clipboard-button/copy-to-clipboard-button.l10n.ts
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/copy-to-clipboard-button/copy-to-clipboard-button.l10n.ts]: Bad idea to do this without CI, imho. It requires some refactoring. https://github.com/vmware/versatile-data-kit/issues/1641

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts]: I haven't commented anything out. https://gitlab.eng.vmware.com/taurus/shared-components/-/blob/person/ikoleva/internal-components-added-cleanup-theme-reduced/projects/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts#L64

File: projects/frontend/shared-components/gui/projects/documentation-ui/.browserslistrc
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/documentation-ui/.browserslistrc:1]: Sounds reasonable.

File: projects/frontend/shared-components/gui/.npmrc
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/.npmrc:1]: What should we add to the VEP that's related to the flag?

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts
Comment (sabadzhiev) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts]: I would say, please investigate in a timeboxed manner what is the issue with the tests. 

If it get's too heavy/time consuming, in order to unblock us, please add a TODO in the code, and put a follow up ticket in the backlog, that will be taken right after we merge all the changes in.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts
Comment (ivakoleva) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts:11]: A username is not a secret? Also, we already have a bunch of author ids and emails.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/commons/ngx-components/search/search.component.spec.ts]: I fixed the issue by updating the zone.js version as detailed here https://github.com/angular/angular/issues/45476
Please don't leave commented code lying around, it's just confusing.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts:11]: I'll just remove them, cause most likely nobody will contact the author directly for support.

File: projects/frontend/shared-components/gui/.npmrc
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/.npmrc:1]: https://github.com/vmware/versatile-data-kit/issues/1643

File: projects/frontend/shared-components/gui/angular.json
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/angular.json:124]: https://github.com/vmware/versatile-data-kit/issues/1644

File: projects/frontend/shared-components/gui/projects/documentation-ui/.browserslistrc
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/projects/documentation-ui/.browserslistrc:1]: https://github.com/vmware/versatile-data-kit/issues/1643

File: projects/frontend/shared-components/gui/package.json
Comment (DeltaMichael) [reply projects/frontend/shared-components/gui/package.json]: I'll set the version to `0.0.1` following your example here https://github.com/vmware/versatile-data-kit/pull/1626/files#diff-b2c4b01f8643ec76b11a2a2dcfc3903d534023cda3ee4b90dc1f8719b9b43dd4

**Comment by ivakoleva** [comment]
Please link the tickets created to this PR

**Comment by DeltaMichael** [comment]
Linking the issues will just close them once we merge. I've put them in the description instead.

File: projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts
Comment (sabadzhiev) [reply projects/frontend/shared-components/gui/projects/shared/src/lib/common/interfaces/comparable.interface.ts]: Iva, generally none of these stuff is a secret, but these brings no meaningful value outside of the context of VMW systems :). 

If I'm an external user, I guess I'll have hard times understanding who is 'gorankokin'. 

If we want to keep them, may be we should enhance the author information with name/e-mail potentially etc. stuff, which I believe is easier just to move out. 

But this is my personal opinion :), so I'll will resolve this thread and will leave you the decision how to proceed here. If you like to leave these as is, I do not mind.

**Review by sabadzhiev — APPROVED** [review]

**Pull request merged**
